modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php

```diff
@@ -0,0 +1,308 @@
+          'unsigned' => FALSE,
+          'alias' => 'st',
+        ],
+      ],
+      TRUE
+    );
+
+    // Define two different workflows.
+    $workflow_base = $this->createMock(WorkflowInterface::class);
+    $workflow_base->expects($this->any())
+      ->method('get')
+      ->with('type_settings')
+      ->willReturn(
+        [
+          'states' =>
+            [
+              'draft' => [
+                'published' => 0,
+              ],
+              'published' => [
+                'published' => 1,
+              ],
+            ],
+        ]
+      );
+
+    $workflow_alternative = $this->createMock(WorkflowInterface::class);
+    $workflow_alternative->expects($this->any())
+      ->method('get')
+      ->with('type_settings')
+      ->willReturn(
+        [
+          'states' =>
+            [
+              'invalid' => [
+                'published' => 0,
+              ],
+              'valid' => [
+                'published' => 1,
+              ],
+            ],
+        ]
+      );
+
+    $configEntityTypeInterface = $this->createMock(ConfigEntityTypeInterface::class);
+    $configEntityTypeInterface->expects($this->any())
+      ->method('getConfigPrefix')
+      ->willReturn('workflows.workflow');
+
+    $storage = $this->createMock(EntityStorageInterface::class);
+    $storage->expects($this->any())
+      ->method('load')
+      ->willReturnMap(
+        [
+          ['oe_corporate_workflow', $workflow_base],
+          ['test', $workflow_alternative],
+        ]
+      );
+
+    $this->entityTypeManager = $this->createMock(EntityTypeManagerInterface::class);
+    $this->entityTypeManager->expects($this->any())
+      ->method('getDefinition')
+      ->with('workflow')
+      ->willReturn($configEntityTypeInterface);
+    $this->entityTypeManager->expects($this->any())
+      ->method('getStorage')
+      ->with('workflow')
+      ->willReturn($storage);
+
+    $this->configManager = $this->createMock(ConfigManagerInterface::class);
+    $this->configManager->expects($this->any())
+      ->method('getEntityTypeIdByName')
+      ->with($this->logicalOr('workflows.workflow.oe_corporate_workflow', 'workflows.workflow.test'))
+      ->willReturn('workflow');
+  }
+
+  /**
+   * Test de base case without any special configuration.
+   *
+   * @throws \Drupal\migrate\MigrateException
+   * @throws \Exception
+   */
+  public function testTransform() {
+    $this->initializePlugin();
+
+    $value = 'published';
+    $statusDestination = $this->plugin->transform($value, $this->migrateExecutable, $this->row, $this->destinationProperty);
+    $this->assertEquals($statusDestination, $value);
+
+    $value = 'draft';
+    $this->row->setSourceProperty('status', 0);
+    $this->row->setSourceProperty('name', $value);
+    $statusDestination = $this->plugin->transform($value, $this->migrateExecutable, $this->row, $this->destinationProperty);
+    $this->assertEquals($statusDestination, $value);
```

Comment:
`assertEquals` takes the expected parameters first always.

And always please try to use a string or array representation in the expected, and not a variable. This makes the assertion clear as well as prevents possible false positives due to variable value change. Also, commenting assertions help readability a lot.

```
// The resulting value should be "draft" because that is the default plugin configuration that matches with a status of 0.
$this->assertEquals('draft', $statusDestination);
```

Do this everywhere please.

Reply:
You're totally right. Changed in all places.